Feature/add setting start end acceptable latetime

phillipet opened the pull request.

phillipet committed:
update map authRange in admin mode

phillipet committed:
add startTime, endTime, acceptableLateTime to service, model, notifier

phillipet committed:
add startTime, endTime, acceptableLateTime to UI

phillipet committed:
fix save office model

phillipet committed:
add DateTimeField only show time

phillipet committed:
Merge branch 'dev' into feature/add-setting-start-end-latereason

phillipet committed:
Merge remote-tracking branch 'upstream/dev' into feature/add-setting-start-end-latereason

phillipet committed:
fix return current value time if use cancel time picker

phillipet committed:
format code

hemvanh commented on lib/ui/widgets/office_settings.dart:
```diff
@@ -101,6 +120,32 @@ class OfficeSettings extends StatelessWidget {
+              controller: starTimeController,
+              style: TextStyle(color: notifier.busy ? Colors.grey : Colors.black),
+              format: DateFormat.Hm(),
+              onShowPicker: showPicker),
```
showPicker,),

hemvanh commented on lib/core/models/office.dart:
```diff
@@ -6,6 +6,9 @@ class Office {
   double lng;
   double authRange;
   String dateIpServiceUrl;
+  String startTime;
```
use Datetime datatype

hemvanh commented on lib/ui/widgets/office_settings.dart:
```diff
@@ -123,6 +168,9 @@ class OfficeSettings extends StatelessWidget {
                             lngController.text,
                             authRangeController.text,
                             dateIpServiceUrlController.text,
+                            starTimeController.text,
```
save to firebase as proper .toIso8601String()
https://api.dart.dev/stable/2.7.1/dart-core/DateTime/toIso8601String.html

hemvanh commented on lib/core/models/office.dart:
```diff
@@ -6,6 +6,9 @@ class Office {
   String dateIpServiceUrl;
+  String startTime;
+  String endTime;
+  String acceptableLateTime;
```
-> int

phillipet committed:
change data type of startTime, endTime, acceptableLateTime

(firebase->model->service->notifier-service)
service received model type
notifer follow by UI logic

phillipet committed:
format code

phillipet committed:
format code
